src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.Prerendering.cs

```diff
@@ -229,9 +229,29 @@ async Task Execute()
                 // Clear all pending work.
                 _nonStreamingPendingTasks.Clear();
 
-                // new work might be added before we check again as a result of waiting for all
-                // the child components to finish executing SetParametersAsync
-                await pendingWork;
+                try
+                {
+                    // new work might be added before we check again as a result of waiting for all
+                    // the child components to finish executing SetParametersAsync
+                    await pendingWork;
+                }
+                catch (Exception ex)
```

Comment:
Why not catch `NavigationException` directly here like everywhere else?

Reply:
True, that would be neater